### DISCUSSION TITLE
Vanilla Extract - Sprinkles (CSS in JS output to static CSS)

### DISCUSSION THREAD

**Discussion by reramjiawan** [remix-run/remix#2567] [Proposals]
https://remix.run/docs/en/v1/guides/styling#css-ecosystem-and-performance

> We are still researching how best to support, and be supported by, the various styling libraries without sacrificing the user's network tab or creating a maintenance burden for Remix.

I'd suggest to have a look at https://vanilla-extract.style/. It's from the author of CSS Modules [markdalgleish](https://twitter.com/markdalgleish).

What it boils down to is that you can have your CSS in JS solution and generate CSS files as output.

Thanks for creating Remix.

**Comment by reramjiawan** [comment]
Addition: https://linaria.dev/ Same concept.

**Comment by machour** [comment]
Mark actually published a demo today with vanilla extract & remix, if you want to have a look: https://twitter.com/markdalgleish/status/1565085790744760320

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
